documentation/specs/Drawer.md

```diff
@@ -0,0 +1,210 @@
+
+### Prior Art
+
+- [Paste `<SideModal />`](https://paste.twilio.design/components/side-modal)
+- [Aria `<Dialog />`](https://react-spectrum.adobe.com/react-aria/Dialog.html)
+
+---
+
+## Design
+
+`Drawer` will use `useDialog` and `useModalOverlay` from `react-aria` to provide the foundation for an accessible dialog.
+
+The component design was inspired by Aria's Dialog component and Twilio's SideModal component.
+
+`Drawer` will manage its own state by default but can be controlled if the consumer opts in.
+
+`Drawer` will be a compound component consistenting of `Drawer`, `Drawer.Header`, and `Drawer.Body`.
+
+While the above components allow for any design, `Drawer` includes `Drawer.Banner`, `Drawer.BanneredContentArea`, `Drawer.StandaloneContentArea`, and `Drawer.CloseButton` for using preset Drawer patterns.
+
+`Drawer` should be attached to a focusable trigger element such as a `Button` through the `Drawer.Trigger` component. This ensures the trigger and modal are accessible. In the event that a focusable element can't be used, `DrawerContainer` can be used for custom triggering.
+
+`Drawer.Trigger` must contain exactly two direct children. The first child must be a focusable trigger such as `Button`. The second child must be either a `Drawer` or a render function that returns a `Drawer`. If using a render function, a `close` argument will be passed to allow for programmatically closing the `Drawer`. This pattern is adopted from the suggested patterns of React Aria and React Spectrum.
+
+`Drawer` will use React CSS Transition Group to slide the dialog in and out of the screen.
+
+### API
+
+```ts
+type DrawerTriggerProps = {
+  /**
+   * Content of drawer trigger. Must be exactly two elements.
+   */
+  children: [ReactElement, CloseableModalElement | ReactElement];
+
+  /**
+   * Whether the drawer is open by default (uncontrolled).
+   */
+  defaultOpen?: boolean;
+
+  /**
+   * Whether or not the drawer can be dismissed.
+   */
+  isDismissable?: boolean;
+
+  /**
+   * Whether the drawer is open by default (controlled).
+   */
+  isOpen?: boolean;
+
+  /**
+   * Handler that is called when the overlay's open state changes.
+   */
+  onOpenChange?: (isOpen: boolean) => void;
+};
+
+type DrawerProps = {
+  /**
+   * Content of the drawer.
+   */
+  children: ReactNode;
+};
+
+type DrawerHeaderProps = {
+  /**
+   * Drawer header content.
+   */
+  children: string;
+};
+
+type DrawerBannerProps = {
+  /**
+   * Drawer header banner content.
+   */
+  children: ReactNode;
+};
+
+type DrawerBodyProps = {
+  /**
+   * Drawer body content.
+   * This provides a scrollable region without any predefined padding.
+   */
+  children: ReactNode;
+};
+
+type DrawerBanneredContentAreaProps = {
+  /**
+   * Content area for a drawer with a banner.
+   */
+  children: ReactNode;
+};
+
+type DrawerStandaloneContentAreaProps = {
+  /**
+   * Content area for a drawer without a banner (standalone).
+   */
+  children: ReactNode;
+};
```

Comment:
open to suggestions on naming and concepts.

the idea is to provide some built-ins for certain drawer content areas.

`Drawer.BanneredContentArea`:

<img width="378" alt="image" src="https://github.com/user-attachments/assets/7966f6b2-cc4c-4d69-aeca-3b83fe6afeda">

`Drawer.StandaloneContentArea`:

<img width="451" alt="image" src="https://github.com/user-attachments/assets/d8f7731f-0740-40e2-9495-7af10629229b">

this is admittedly a bit wordy but it allows for any flexibility for future `Drawer` designs. an alternative might be exposing a general `Drawer.ContentArea` with `padding` props.

Reply:
im good with the the naming you chose, its sensible to me